pkg/resource/settings/download.go

```diff
@@ -50,29 +49,45 @@ type schema struct {
 	ownerBasedAccessControl *bool
 }
 
-func Download(ctx context.Context, client client.SettingsClient, projectName string, filters Filters, schemaIDs ...config.SettingsType) (project.ConfigsPerType, error) {
-	if len(schemaIDs) == 0 {
-		return downloadAll(ctx, client, projectName, filters)
+type Source interface {
+	ListSchemas(context.Context) (dtclient.SchemaList, error)
+	List(context.Context, string, dtclient.ListSettingsOptions) ([]dtclient.DownloadSettingsObject, error)
+	GetPermission(context.Context, string) (dtclient.PermissionObject, error)
+}
+
+type API struct {
+	settingsSource Source
+	filters        Filters
+	schemaIDs      []config.SettingsType
+}
+
+func NewAPI(settingsSource Source, filters Filters, schemaIDs ...config.SettingsType) *API {
+	return &API{settingsSource, filters, schemaIDs}
```

Comment:
Because we aren't using `config.SettingsType` in any way except mapping it to a different data-structure, would it make sense to put the settingsType => string mapping in here instead of Download?